test/WebJobs.Script.Tests/Workers/Http/HttpWorkerFunctionProviderTests.cs

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Workers.Http;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Moq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Workers.Http
+{
+    public class HttpWorkerFunctionProviderTests
+    {
+        private static IOptions<HttpWorkerOptions> CreateHttpWorkerOptions(IEnumerable<HttpWorkerRoute> routes = null, bool customRoutesEnabled = true, bool includeHttpSection = true)
+        {
+            var options = new HttpWorkerOptions
+            {
+                CustomRoutesEnabled = customRoutesEnabled
+            };
+
+            if (includeHttpSection)
+            {
+                options.Http = new CustomHandlerHttpOptions
+                {
+                    Routes = routes
+                };
+            }
+
+            return Options.Create(options);
+        }
+
+        private static TestOptionsMonitor<LanguageWorkerOptions> CreateLanguageWorkerOptions()
+        {
+            var lang = new LanguageWorkerOptions
+            {
+                WorkerConfigs = []
+            };
+            return new TestOptionsMonitor<LanguageWorkerOptions>(lang);
+        }
+
+        private static HttpWorkerFunctionProvider CreateProvider(
+            IOptions<HttpWorkerOptions> httpOptions,
+            IHostFunctionMetadataProvider hostMetadataProvider)
+        {
+            return new HttpWorkerFunctionProvider(
+                httpOptions,
+                CreateLanguageWorkerOptions(),
+                hostMetadataProvider,
+                new NullLogger<HttpWorkerFunctionProvider>());
+        }
+
+        [Fact]
+        public async Task GetFunctionMetadataAsync_NoHttpSection_ReturnsEmpty()
+        {
+            var hostMeta = new Mock<IHostFunctionMetadataProvider>(MockBehavior.Strict);
+            var provider = CreateProvider(
+                CreateHttpWorkerOptions(routes: null, customRoutesEnabled: true, includeHttpSection: false),
+                hostMeta.Object);
+
+            var result = await provider.GetFunctionMetadataAsync();
+
+            Assert.Empty(result);
```

Comment:
If you are interested, I much prefer `AwesomeAssertions` (aka FluentAssertions) over Xunit.Assert.


```suggestion
            result.Should().BeEmpty();
```